lib/daru/view/adapters/googlecharts/display.rb

```diff
@@ -117,13 +142,43 @@ def append_data(data)
       "\n  \t\tdataTable: data_table,"
     end
 
+    # @param element_id [String] The ID of the DIV element that the Google
+    #   Chart should be rendered in
+    # @return [String] unique function name to save the chart
+    def save_chart_function_name(element_id)
+      "saveChart_#{element_id.tr('-', '_')}"
+    end
+
+    # @param (see #draw_js_chart_editor)
+    # @return [String] options of the ChartWrapper
+    def extract_chart_wrapper_options(data, element_id)
+      js = ''
+      js << if is_a?(GoogleVisualr::DataTable)
+              "\n  \t\tchartType: 'Table',"
+            else
+              "\n  \t\tchartType: '#{chart_name}',"
+            end
+      js << append_data(data)
+      js << "\n  \t\toptions: #{js_parameters(@options)},"
+      js << "\n  \t\tcontainerId: '#{element_id}',"
+      js << "\n  \t\tview: #{extract_option_view}"
+      js
+    end
+
     # So that it can be used in ChartEditor also
     #
     # @return [String] Returns string to draw the Chartwrapper and '' otherwise
-    def draw_wrapper
-      return "\n  \twrapper.draw();" if
+    def draw_wrapper(element_id)
+      return "\n  \twrapper_#{element_id.tr('-', '_')}.draw();" if
```

Comment:
Just want to know, what if we don't provide `element_id `?

Reply:
If we use only the `wrapper` (without id) then the chart editor will work only on the last chart. That is, edit button (of any chart) will show the dialog box for the last chart only.